go/common/cgroups_rootless.go

```diff
@@ -0,0 +1,31 @@
+package common
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func CgroupPath() string {
+	// Attempt to execute in rootless mode
+	cmd := exec.Command("systemctl", "show", "--user", "--property=ControlGroup")
+	output, err := cmd.Output()
+	if err == nil {
+		parts := strings.SplitN(strings.TrimSpace(string(output)), "=", 2)
+		if len(parts) == 2 {
+			path := filepath.Join("/sys/fs/cgroup", parts[1])
+			if st, err := os.Stat(path); err == nil && st.IsDir() {
+				return path
+			}
+		}
+	}
+
+	// Use default path if run as root
+	if os.Getuid() == 0 {
+		return "/sys/fs/cgroup"
+	}
+
+	panic(fmt.Errorf("systemd user cgroup delegation not available - cannot run rootless"))
```

Comment:
I think instead of trying to integrate with systemd directly, I think it is more flexible if we let users configure the root cgroup to use.  We can provide instructions for users to run "systemctl show --user --property=ControlGroup" and find this location and configure their OL accordingly.